core/drivers/pl022_spi.c

```diff
@@ -333,10 +259,11 @@ static void pl022_sanity_check(struct pl022_data *pd)
 {
 	assert(pd);
 	assert(pd->chip.ops);
-	assert(pd->gpio);
-	assert(pd->gpio->ops);
-	assert(pd->base);
-	assert(pd->cs_gpio_base);
+	/* pd->cs_data.cs_cb CAN be NULL so no assert */
```

Comment:
Do you think `pd->cs_data.cs_cb` can really be `NULL` if `pd->b_cs_callback == true` ?
Is it a way to specify there is not Chip Select required ?

If so, I think you should explicitly state this in the structure definition, and I think it's a bit... ugly too. I would prefer to replace `b_cs_callback` with a enumerate field, like `cs = CS_GPIO` or `cs = CS_CALLBACK` or `cs = CS_NONE`.